src/vis/bar/utils.ts

```diff
@@ -0,0 +1,240 @@
+import {merge} from 'lodash';
+import {CategoricalColumn, ColumnInfo, EColumnTypes, ESupportedPlotlyVis, IVisConfig, NumericalColumn, Scales} from '../interfaces';
+import {PlotlyInfo, PlotlyData} from '../interfaces';
+import {getCol} from '../sidebar/utils';
+
+export enum EBarDisplayType {
+    DEFAULT = 'Default',
+    NORMALIZED = 'Normalized',
+}
+
+export enum EBarDirection {
+    VERTICAL = 'Vertical',
+    HORIZONTAL = 'Horizontal',
+}
+
+export enum EViolinOverlay {
+    NONE = 'None',
+    STRIP = 'Strip',
+    BOX = 'Box'
+}
+
+export enum EBarGroupingType {
+    STACK = 'Stacked',
+    GROUP = 'Grouped',
+}
+
+export function isBar(s: IVisConfig): s is IBarConfig {
+    return s.type === ESupportedPlotlyVis.BAR;
+}
+
+export interface IBarConfig {
+    type: ESupportedPlotlyVis.BAR;
+    multiples: ColumnInfo | null;
+    group: ColumnInfo | null;
+    direction: EBarDirection;
+    display: EBarDisplayType;
+    groupType: EBarGroupingType;
+    numColumnsSelected: ColumnInfo[];
+    catColumnsSelected: ColumnInfo[];
+}
+
+const defaultConfig: IBarConfig = {
+    type: ESupportedPlotlyVis.BAR,
+    numColumnsSelected: [],
+    catColumnsSelected: [],
+    group: null,
+    groupType: EBarGroupingType.STACK,
+    multiples: null,
+    display: EBarDisplayType.DEFAULT,
+    direction: EBarDirection.VERTICAL,
+};
+
+export function barMergeDefaultConfig(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig
+): IVisConfig {
+    const merged = merge({}, defaultConfig, config);
+
+    const catCols = columns.filter((c) => c.type === EColumnTypes.CATEGORICAL);
+
+    if(merged.catColumnsSelected.length === 0 && catCols.length > 0) {
+        merged.catColumnsSelected.push(catCols[catCols.length - 1].info);
+    }
+
+    return merged;
+}
+
+export function createBarTraces(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig,
+    scales: Scales,
+): PlotlyInfo {
+    let counter = 1;
+
+    if(!config.catColumnsSelected) {
+        return {
+            plots: [],
+            legendPlots: [],
+            rows: 0,
+            cols: 0,
+            errorMessage: 'To create a Bar Chart, please select at least 1 categorical column.',
+            formList: ['groupBy', 'barMultiplesBy', 'barDirection', 'barGroupType', 'barNormalized']
+        };
+    }
+
+    const catCols: CategoricalColumn[] = columns.filter((c) => config.catColumnsSelected.filter((d) => c.info.id === d.id).length > 0 && c.type === EColumnTypes.CATEGORICAL) as CategoricalColumn[];
+    const vertFlag = config.direction === EBarDirection.VERTICAL;
+    const normalizedFlag = config.display === EBarDisplayType.NORMALIZED;
+    const plots: PlotlyData[] = [];
+
+    if(catCols.length > 0) {
+        const catCurr = catCols[0];
+        if(config.group && config.multiples) {
```

Comment:
Maybe extract the different if-else blocks into separate functions to make the if/else here more readable.

```suggestion
        if(config.group && config.multiples) {
        	plots = lotsForGroupAndMultiple(columns, config);
    	} else if(...) {
```

Reply:
agreed, though i'd love to find some way in the future to reduce the crazy code duplication here, but plotly makes it hard. Separated into different functions for now.